### PR TITLE
Enable ref as valid prop

### PR DESCRIPTION
**Pull request by davesnx**
Depends on #859, that depends on #846 

Allows to have a prop with `~ref`

---

This PR also adds the cleanup function on `ref` as a new constructor: added another "constructor" instead of changing the current callbackRef to avoid a breaking change, we might add deprecations and further cleanup in the future versions to keep those constructors to the minimum

### CODE REVIEW

File: test/Ref__test.re [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  test("can be passed as prop to a component", () => {
+    let domRef = React.createRef();
+    let container =
+      ReactTestingLibrary.render(
+        <Button_with_ref ref={ReactDOM.Ref.domRef(domRef)}>
+          {React.string("Click me")}
+        </Button_with_ref>,
+      );
+    let button = getByRole("FancyButton", container);
+    expect(button->innerHTML)->toBe("Click me");
```
Comment (jchavarri) [inline test/Ref__test.re:27]: Shouldn't we test here something about the `ref`? This test just shows that `Button_with_ref` is rendering ok.

**Review by jchavarri — APPROVED** [review]

File: test/Ref__test.re (hunk 1 above)
Comment (davesnx) [reply test/Ref__test.re:27]: Sure, added another assertion but didn't want to go to clicks and other events to keep the test simple and fast

Comment (jchavarri) [reply test/Ref__test.re:27]: Thanks. (I think the expect in line 27 can be removed, because it doesn't test anything related to the ref)

**Comment by anmonteiro** [comment]
context: https://vercel.com/blog/whats-new-in-react-19#ref-as-a-prop

**Pull request merged**